packages/go_router_builder/lib/src/path_utils.dart

```diff
@@ -4,70 +4,29 @@
 
 final RegExp _parameterRegExp = RegExp(r':(\w+)(\((?:\\.|[^\\()])+\))?');
 
-/// Converts a [pattern] such as `/user/:id` into [RegExp].
-///
-/// The path parameters can be specified by prefixing them with `:`. The
-/// `parameters` are used for storing path parameter names.
+/// Extracts the path parameters from a [pattern] such as `/user/:id`.
 ///
+/// The path parameters can be specified by prefixing them with `:`.
 ///
 /// For example:
 ///
-///  `pattern` = `/user/:id/book/:bookId`
-///
-///  The `parameters` would contain `['id', 'bookId']` as a result of calling
-///  this method.
-///
-/// To extract the path parameter values from a [RegExpMatch], pass the
-/// [RegExpMatch] into [extractPathParameters] with the `parameters` that are
-/// used for generating the [RegExp].
-RegExp patternToRegExp(String pattern, List<String> parameters) {
-  final StringBuffer buffer = StringBuffer('^');
-  int start = 0;
-  for (final RegExpMatch match in _parameterRegExp.allMatches(pattern)) {
-    if (match.start > start) {
-      buffer.write(RegExp.escape(pattern.substring(start, match.start)));
-    }
-    final String name = match[1]!;
-    final String? optionalPattern = match[2];
-    final String regex = optionalPattern != null
-        ? _escapeGroup(optionalPattern, name)
-        : '(?<$name>[^/]+)';
-    buffer.write(regex);
-    parameters.add(name);
-    start = match.end;
-  }
-
-  if (start < pattern.length) {
-    buffer.write(RegExp.escape(pattern.substring(start)));
-  }
-
-  if (!pattern.endsWith('/')) {
-    buffer.write(r'(?=/|$)');
-  }
-  return RegExp(buffer.toString(), caseSensitive: false);
-}
```

Comment:
In this commit ([refactor: Simplify path utils methods to only keep what is used](https://github.com/flutter/packages/pull/4524/commits/205da049bc308c5643ad6fdf62b19e26e8a0bff0)),

I'm only deleting and simplifying `patternToRegExp`. We don't need to used the returnd `RegExp`. So instead I'm returning the `parameters` (and I don't need the parameter `parameters` anymore)